src/components/TooltipNew/GuideTooltip.js

```diff
@@ -0,0 +1,272 @@
+    width: '100%',
+    height: `${gap.top}px`
+  };
+  const overlayBottom = {
+    top: `${gap.bottom}px`,
+    left: '0',
+    width: '100%',
+    height: `calc(100% - ${gap.bottom}px)`
+  };
+
+  return (
+    <React.Fragment>
+      <div
+        className={cx({
+          [styles[`${baseClass}__overlay`]]: true,
+          [styles[`${baseClass}__overlay--visible`]]: isVisible,
+          [styles[`${baseClass}__overlay--slide`]]: slide
+        })}
+        style={overlayLeft}
+      />
+      <div
+        className={cx({
+          [styles[`${baseClass}__overlay`]]: true,
+          [styles[`${baseClass}__overlay--visible`]]: isVisible,
+          [styles[`${baseClass}__overlay--slide`]]: slide
+        })}
+        style={overlayTop}
+      />
+      <div
+        className={cx({
+          [styles[`${baseClass}__overlay`]]: true,
+          [styles[`${baseClass}__overlay--visible`]]: isVisible,
+          [styles[`${baseClass}__overlay--slide`]]: slide
+        })}
+        style={overlayRight}
+      />
+      <div
+        className={cx({
+          [styles[`${baseClass}__overlay`]]: true,
+          [styles[`${baseClass}__overlay--visible`]]: isVisible,
+          [styles[`${baseClass}__overlay--slide`]]: slide
+        })}
+        style={overlayBottom}
+      />
+    </React.Fragment>
+  );
+};
+
+GapOverlay.propTypes = {
+  gap: PropTypes.shape({
+    top: PropTypes.number,
+    left: PropTypes.number,
+    bottom: PropTypes.number,
+    right: PropTypes.number,
+    width: PropTypes.number,
+    height: PropTypes.number
+  }),
+  isVisible: PropTypes.bool,
+  slide: PropTypes.bool
+};
+
+class GuideTooltip extends React.PureComponent {
+  static getDerivedStateFromProps(props, state) {
+    if (props.isVisible) {
+      if (!state.lastElement) {
+        // this is the first time the tooltip is shown - don't slide from the (0, 0) origin
+        return {
+          shouldSlide: false,
+          lastElement: props.element
+        };
+      }
+
+      if (props.element !== state.lastElement) {
+        // the element has changed - slide to the next one
+        return {
+          shouldSlide: true,
+          lastElement: props.element
+        };
+      }
+
+      // window was scrolled or any other change has happened - don't slide
+      return {
+        shouldSlide: false
+      };
+    }
+
+    // reset the element when the tooltip hides
+    return {
+      lastElement: null
+    };
+  }
+
+  constructor(props) {
+    super(props);
+
+    // we don't use the following state params but we still need to re-render when the window changes
+    /* eslint-disable react/no-unused-state */
+    this.state = {
+      windowWidth: 0,
+      windowHeight: 0,
+      windowScrollX: 0,
+      windowScrollY: 0
+    };
+    /* eslint-enable react/no-unused-state */
+
+    this.handleDocumentChange = this.handleDocumentChange.bind(this);
+  }
+
+  componentDidMount() {
+    window.addEventListener('resize', this.handleDocumentChange);
```

Comment:
shouldn't the handlers be throttled/debounced?

Reply:
I believe it would give the impression of a lag when the window is scrolling - the spotlight would try to "catch-up" with the element. It might be a subject to discuss but I don't think the calculations are that much resource consuming. I don't think it's worth it.

Reply:
I think we should throttle it to keep 60fps. It would limit one event per 16ms, but keep scroll smooth.

Reply:
I added throttling 16 ms